### PR TITLE
Feat: Update @safe-global/safe-apps-sdk to ^8.0.0

### LINKED ISSUE #2337: Update `@safe-global/safe-apps-sdk` to the latest version
## What is the feature about
Updating the safe apps SDK to the latest version (8.0.0 when creating the issue). It should fix bugs like:
- https://github.com/safe-global/safe-apps-sdk/issues/480

## The list of requirements
- Update safe apps SDK to the latest version

## Links
- https://www.npmjs.com/package/@safe-global/safe-apps-sdk

### PR DESCRIPTION
**Pull request by mmv08**
## What it solves

Resolves #2337 

## How this PR fixes it 
By updating the safe apps SDK to the latest version. I also had to fix a couple of linter errors and add `TextEncoder` to the global env in jest because it was required by `viem` but not included with JSDom.

## How to test it
Open a test safe app, and try to execute a couple of transactions and message signings.

## Checklist
* [ ] I've tested the branch on mobile 📱
* [x] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

**Review by schmanu — DISMISSED** [review]
Something is not right. 
The deployment failed and all unit tests that import from the safe-apps-sdk fail.

File: src/components/safe-apps/types.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ const FEATURES = [
   'speaker-selection',
 ]
 
-export type AllowedFeatures = typeof FEATURES[number]
+export type AllowedFeatures = (typeof FEATURES)[number]
```
Comment (mmv08) [inline src/components/safe-apps/types.ts:43]: This threw a linter error without the braces

File: src/services/pairing/module.ts [hunk 2]
```diff
@@ -56,11 +56,11 @@ const pairingModule = (): WalletInit => {
...
           public disconnect: EIP1193Provider['disconnect']
           // @ts-expect-error - 'emit' does not exist on `typeof EventEmitter`
-          public emit: typeof EventEmitter['emit']
+          public emit: (typeof EventEmitter)['emit']
           // @ts-expect-error - 'on' does not exist on `typeof EventEmitter`
-          public on: typeof EventEmitter['on']
+          public on: (typeof EventEmitter)['on']
           // @ts-expect-error - 'removeListener' does not exist on `typeof EventEmitter`
-          public removeListener: typeof EventEmitter['removeListener']
+          public removeListener: (typeof EventEmitter)['removeListener']
```
Comment (mmv08) [inline src/services/pairing/module.ts:63]: This threw a linter error without the braces

**Review by katspaugh — APPROVED** [review]

**Comment by katspaugh** [comment]
@mmv08 please solve the git conflicts, I've moved it to QA in the meanwhile.

**Review by iamacook — APPROVED** [review]
There's a merge conflict.

**Comment by mmv08** [comment]
sorted

**Comment by mmv08** [comment]
The same tests fail in all the other PRs, so it isn't affected by my changes. Correct?

**Comment by katspaugh** [comment]
Yes, the tests have unfortunately been flaky recently because of Goerli (supposedly).

**Comment by francovenica** [comment]
Works for me.
Note: A lot of the apps don't work in test envs.
I tried with Cow swap, Aave and a test app we use to generate custom messages. Still it would be great if we can ask the person who reported it to give it a try again since idk what app can requiere this "undefined" value

**Comment by mmv08** [comment]
I've realised that the `undefined` issue might be something different - in the transaction details parsing from the safe apps SDK message on the wallet side. But it is still worth updating the safe-apps-sdk to the latest version as it should improve the bundle size

**Review by schmanu — APPROVED** [review]

**Comment by mmv08** [comment]
I'll leave up to you to merge this since I might be slightly unaware of the release processes/plans

**Comment by katspaugh** [comment]
We'll include it in the release after the upcoming one if you don't mind.

**Pull request merged**